File: puttyknife/spi/AnimationState.c
#include <spi/AnimationState.h>

spAnimationState *spiAnimationState_create(spSkeletonData *skeletonData)
{

    return spAnimationState_create(spAnimationStateData_create(skeletonData));
}

void spiAnimationState_dispose(spAnimationState *self)
{
    spAnimationStateData_dispose(self->data);
    spAnimationState_dispose(self);
}

spTrackEntry *spiAnimationState_addAnimationByName(spAnimationState *self, int trackIndex, const char *animationName,
                                                   int /*bool*/ loop, float delay)
{
    return spAnimationState_addAnimationByName(self, trackIndex, animationName, loop, delay);
}

void spiAnimationState_update(spAnimationState *self, float delta)
{
    spAnimationState_update(self, delta);
}

int spiAnimationState_apply(spAnimationState *self, spSkeleton *skeleton)
{
    return spAnimationState_apply(self, skeleton);
}

File: script/spine/AnimationState.ts
import { pk } from "puttyknife";
import { SkeletonData } from "./SkeletonData.js";

export class AnimationState {
    readonly pointer: number;

    constructor(data: SkeletonData) {
        this.pointer = pk.fn.spiAnimationState_create(data.pointer);
    }

    addAnimationByName(trackIndex: number, animationName: string, loop: boolean, delay: number) {
        const animationNamePtr = pk.heap.addString(animationName);
        pk.fn.spiAnimationState_addAnimationByName(this.pointer, trackIndex, animationNamePtr, loop ? 1 : 0, delay);
        pk.heap.delString(animationNamePtr);
    }

    update(dt: number): void {
        pk.fn.spiAnimationState_update(this.pointer, dt);
    }
}